kernel/libc/koslib/dbglog.c

```diff
@@ -21,9 +21,14 @@ static spinlock_t mutex = SPINLOCK_INITIALIZER;
    DBG_KDEBUG to see everything. DBG_INFO is generally a decent level. */
 int dbglog_level = DBG_KDEBUG;
 
-/* Set debug level */
+/* Set debug level if DBG_DYNLOG is enabled */
 void dbglog_set_level(int level) {
+#ifdef DBG_DYNLOG
     dbglog_level = level;
+#else
+    dbgio_write_str("dbglog: Define DBG_DYNLOG to change log level.\n \
```

Comment:
```suggestion
#include <assert.h>

    static_assert(false, "dbglog: Define DBG_DYNLOG to change log level.\n \
```
Now that we're on a revision of C from this century, we can make checks like this compile-time. I would move it outside of the function too to global scope.

Reply:
Oh, sure. I was basically just having it as the softest warning possible but still building. The check could always be compile time and just using `#warning` too. Don't know if anyone's actually ever used the function anyways (maybe in lieu of figuring out the defines and setting it at the start of main).